druid/src/core.rs

```diff
@@ -106,6 +106,25 @@ pub(crate) struct WidgetState {
     pub(crate) viewport_offset: Vec2,
 
     // TODO: consider using bitflags for the booleans.
+    // Any child of this widget changed the disabled state and should either receive
+    // LifeCycle::DisabledChanged or InternalLifeCycle::RouteDisabledChanged
+    pub(crate) children_disabled_changed: bool,
+
+    // Any of our ancestors are disabled. Therefore we are also disabled.
+    pub(crate) ancestor_disabled: bool,
+
+    // explicitly disabled through the usage of EventCtx::set_disabled(true) on this widget.
+    pub(crate) is_explicitly_disabled: bool,
+
+    // A buffer for the value of EventCtx::set_disabled() before this widget receives
+    // LifeCycle::DisabledChanged or InternalLifeCycle::RouteDisabledChanged
```

Comment:
```suggestion
    // `true` if this widget has been explicitly disabled, but has not yet seen one of
    // LifeCycle::DisabledChanged or InternalLifeCycle::RouteDisabledChanged
```

(Is  this true?)

Also: it confuses me that this is a bool, since I also expect that we would want to know if a widget is newly *enabled*, e.g. it feels like there's a semantic difference between "nothing has changed" and "I have changed to become not-disabled"; both of these states are current represented by `false`.

I think I understand why this works, but I did find it confusing.